packages/near-api-js/src/account_multisig.ts

```diff
@@ -381,6 +396,61 @@ export class Account2FA extends AccountMultisig {
         });
     }
 
+    /**
+     * Returns `false` if multisig cannot be disabled by calling disable() directly
+     * i.e. account has too many LAKs to delete and re-create as FAKs for a single transaction
+     */
+    async canDisableMultisig() {
+        const keys = await this.get2faLimitedAccessKeys();
+
+        // there are max 4 actions required to disable 2FA on top of the key conversion (2 actions per key)
+        // (100 - 4) / 2 = 48 = LAK_CONVERSION_LIMIT - 2
+        return keys.length <= (LAK_CONVERSION_LIMIT - 2);
+    }
+
+    /**
+     * Converts multisig LAKs (excluding the active key in the wallet, provided here as the
+     * `signingPublicKey` argument) back to FAKs when there are too many key conversions to be
+     * executed within the same transaction.
+     *
+     * Note that this undermines the security provided by 2FA by converting existing multisig LAKs into FAKs
+     * capable of bypassing the 2FA prompt when signing transactions outside the wallet (transactions signed
+     * in the wallet will continue to require 2FA approval). This method should only be used to convert LAKs
+     * as a way to make 2FA disabling possible.
+     *
+     * To be deprecated after 2FA migration; this addresses a specific issue with multisig deployed via wallet.
+     * @param signingPublicKey the public key used to sign transactions in the wallet
+     */
+    async batchConvertKeys(signingPublicKey: string) {
```

Comment:
suggestion: maybe we should name the arg to something more general like `whitelist` or `keysToNotConvert` just so it's clearer that this is the key we're leaving behind and that its not being used to sign anything in the context of this method. It would also not tie us to the wallet scenario.

Reply:
This is a really specific scenario that only near.org wallet implements, but even then I don't see any reason to leave more than the one key you'll use to sign the transaction to disable. Specifying that one key is used to sign the transaction to disable multisig is more intentional and leaves less opportunity for keys to end up in an inconsistent state.